src/webhooks/__tests__/registry.test.ts

```diff
@@ -365,277 +376,129 @@ describe('shopify.webhooks.process', () => {
-          rawBody: (req as any).rawBody,
-          rawRequest: req,
-          rawResponse: res,
-        });
-        res.writeHead(response.statusCode);
-        res.end();
-      } catch (error) {
-        errorThrown = true;
-        res.writeHead(error.response.code);
-        res.end();
-      }
-      expect(errorThrown).toBeFalsy();
-      expect(response.statusCode).toBe(200);
-    });
-
-    await request(app)
+    const response = await request(app)
       .post('/webhooks')
       .set(headers({hmac: hmac(shopify.config.apiSecretKey, rawBody)}))
-      .send(rawBody)
-      .expect(StatusCode.Ok);
+      .send(rawBody);
+
+    expect(response.status).toEqual(StatusCode.Ok);
+    expect(response.body.errorThrown).toBeFalsy();
   });
 
   it('handles lower case headers', async () => {
     shopify.webhooks.addHandler({
       topic: 'PRODUCTS',
-      registryEntry: {
-        path: '/webhooks',
-        webhookHandler: genericWebhookHandler,
-      },
-    });
-    const app = express();
-    app.use(parseRawBody);
-    app.post('/webhooks', async (req, res) => {
-      let errorThrown = false;
-      let response: NormalizedResponse = {
-        statusCode: 200,
-        statusText: 'OK',
-      };
-      try {
-        response = await shopify.webhooks.process({
-          rawBody: (req as any).rawBody,
-          rawRequest: req,
-          rawResponse: res,
-        });
-        res.writeHead(response.statusCode);
-        res.end();
-      } catch (error) {
-        errorThrown = true;
-        res.writeHead(error.response.code);
-        res.end();
-      }
-      expect(errorThrown).toBeFalsy();
-      expect(response.statusCode).toBe(200);
+      ...genericRegistryEntry,
     });
 
-    await request(app)
+    const response = await request(app)
       .post('/webhooks')
       .set(
         headers({
           hmac: hmac(shopify.config.apiSecretKey, rawBody),
           lowercase: true,
         }),
       )
-      .send(rawBody)
-      .expect(StatusCode.Ok);
+      .send(rawBody);
+
+    expect(response.status).toEqual(StatusCode.Ok);
+    expect(response.body.errorThrown).toBeFalsy();
   });
 
   it('handles the request and returns Not Found when topic is not registered', async () => {
     shopify.webhooks.addHandler({
       topic: 'NONSENSE_TOPIC',
-      registryEntry: {
-        path: '/webhooks',
-        webhookHandler: genericWebhookHandler,
-      },
+      ...genericRegistryEntry,
     });
 
-    const app = express();
-    app.use(parseRawBody);
-    app.post('/webhooks', async (req, res) => {
-      let errorThrown = false;
-      try {
-        await shopify.webhooks.process({
-          rawBody: (req as any).rawBody,
-          rawRequest: req,
-          rawResponse: res,
-        });
-      } catch (error) {
-        errorThrown = true;
-        res.writeHead(error.response.code);
-        res.end();
-        expect(error).toBeInstanceOf(ShopifyErrors.InvalidWebhookError);
```

Comment:
Should we also include the exception in the response data, so that we can keep running this check?

Reply:
The check is still being run, just inside the post `/webhooks` handler - https://github.com/Shopify/shopify-api-node/blob/aabbbcc8ef0f606499131de52ca660796c19e6d2/src/webhooks/__tests__/registry.test.ts#L350-L367

(see Ln 363)